code/modules/modular_computers/file_system/program.dm

```diff
@@ -83,32 +83,41 @@
 // Check if the user can run program. Only humans can operate computer. Automatically called in run_program()
 // User has to wear their ID or have it inhand for ID Scan to work.
 // Can also be called manually, with optional parameter being access_to_check to scan the user's ID
-/datum/computer_file/program/proc/can_run(var/mob/living/user, var/loud = 0, var/access_to_check)
+/datum/computer_file/program/proc/can_run(var/mob/living/user, var/loud = 0, var/list/accesses_to_check, var/datum/computer_network/network)
 	if(!requires_access_to_run)
-		return 1
-	// Defaults to required_access
-	if(!access_to_check)
-		access_to_check = required_access
-	if(!access_to_check) // No required_access, allow it.
-		return 1
+		return TRUE
+	// Checks to see if network access is enabled, and then defaults to required_access if not.
+	if(!accesses_to_check)
+		if(network)
+			var/datum/extension/network_device/acl/access_controller = network.access_controller
+			if(access_controller && access_controller.program_control)
+				accesses_to_check = access_controller.get_program_access(filename)
+	
+	if(!length(accesses_to_check))
+		accesses_to_check = required_access
+
+	if(!length(accesses_to_check)) // No required_access, allow it.
+		return TRUE
 
 	// Admin override - allows operation of any computer as aghosted admin, as if you had any required access.
 	if(isghost(user) && check_rights(R_ADMIN, 0, user))
-		return 1
+		return TRUE
 
 	if(!istype(user))
-		return 0
+		return FALSE
 
 	var/obj/item/card/id/I = user.GetIdCard()
 	if(!I)
 		if(loud)
-			to_chat(user, "<span class='notice'>\The [computer] flashes an \"RFID Error - Unable to scan ID\" warning.</span>")
-		return 0
-
-	if(access_to_check in I.access)
-		return 1
-	else if(loud)
-		to_chat(user, "<span class='notice'>\The [computer] flashes an \"Access Denied\" warning.</span>")
+			to_chat(user, SPAN_WARNING("The OS flashes an \"RFID Error - Unable to scan ID\" warning."))
+		return FALSE
+
+	for(var/access in I.access)
+		if(access in accesses_to_check)
+			return TRUE
```

Comment:
Use the existing helper (`has_access`); this will correctly handle AND/OR usage in a way consistent with other access stuff.